src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -61,6 +61,32 @@ public static bool IsZipDeployment(this IEnvironment environment)
                 !string.IsNullOrEmpty(environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteRunFromPackage));
         }
 
+        public static bool IsAppServiceWindowsEnvironment(this IEnvironment environment)
+        {
+            return environment.IsAppServiceEnvironment() &&
+                !(environment.IsLinuxAppServiceEnvironment() || environment.IsLinuxContainerEnvironment());
+        }
+
+        public static bool IsCoreToolsEnvironment(this IEnvironment environment)
+        {
+            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(EnvironmentSettingNames.CoreToolsEnvironment));
+        }
+
+        public static bool IsPersistentStorageAvailable(this IEnvironment environment)
+        {
+            return environment.IsAppServiceWindowsEnvironment() ||
+                environment.IsLinuxAppServiceEnvironmentWithStorage()
+                || environment.IsCoreToolsEnvironment();
+        }
+
+        public static bool IsLinuxAppServiceEnvironmentWithStorage(this IEnvironment environment)
+        {
+            return environment.IsLinuxAppServiceEnvironment() &&
```

Comment:
Is storage optional for LinuxAppServiceEnvironment? Where is IsPersistentStorageAvailable used?

Reply:
Persistent storage is optional in linux. There is some configuration that governs that. I can change it to `withPersistentStorage` But it would be extremely long.